whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/dao/Nrtm4ClientInfoRepository.java

```diff
@@ -52,15 +61,24 @@ SELECT id, source, MAX(version), session_id, type, hostname, created
             GROUP BY source
             """;
         return jdbcSlaveTemplate.query(sql,
-                (rs, rn) -> new NrtmClientVersionInfo(
-                        rs.getLong(1),
-                        rs.getString(2),
-                        rs.getLong(3),
-                        rs.getString(4),
-                        NrtmClientDocumentType.fromValue(rs.getString(5)),
-                        rs.getString(6),
-                        rs.getLong(7)
-                        ), NrtmClientDocumentType.NOTIFICATION.getFileNamePrefix());
+                nrtmClientVersionRowMapper(),
+                NrtmClientDocumentType.NOTIFICATION.getFileNamePrefix());
+    }
+
+    @Nullable
+    public NrtmClientVersionInfo getNrtmLastVersionInfoForDeltasPerSource(final String source){
+        try {
+            return jdbcMasterTemplate.queryForObject("""
+                    SELECT id, source, MAX(version), session_id, type, hostname, created
+                    FROM version_info
+                    WHERE type = ?
+                    AND source = ?
+                    """,
+                    nrtmClientVersionRowMapper(),
+                    NrtmClientDocumentType.DELTA.getFileNamePrefix(), source);
```

Comment:
Do you need to save whether version is delta or snapshot? It should not really matter

Reply:
discussed via stack, we will leave it. The code is still clean